lib/jsonapi/utils/response/formatters.rb

```diff
@@ -14,7 +14,9 @@ def jsonapi_format(records, options = {})
         alias_method :jsonapi_serialize, :jsonapi_format
 
         def jsonapi_format_errors(data)
-          data   = JSONAPI::Utils::Exceptions::ActiveRecord.new(data, @request.resource_klass, context) if data.is_a?(ActiveRecord::Base)
+          if data.is_a?(ActiveRecord::Base)|| data.singleton_class.include?(ActiveModel::Model)
```

Comment:
Can you define this boolean expression in a separate private predicate method? Just to make the line clearer :-)

```
def active_record_obj?
  data.is_a?(ActiveRecord::Base)|| data.singleton_class.include?(ActiveModel::Model)
end
```